newrelic/resource_newrelic_cloud_azure_integrations.go

```diff
@@ -0,0 +1,2628 @@
+	} else if o, n := d.GetChange("storage"); len(n.([]interface{})) < len(o.([]interface{})) {
+		cloudDisableAzureIntegration.AzureStorage = []cloud.CloudDisableAccountIntegrationInput{{LinkedAccountId: linkedAccountID}}
+	}
+
+	if v, ok := d.GetOk("virtual_machine"); ok {
+		cloudAzureIntegration.AzureVirtualmachine = expandCloudAzureIntegrationVirtualMachineInput(v.([]interface{}), linkedAccountID)
+	} else if o, n := d.GetChange("virtual_machine"); len(n.([]interface{})) < len(o.([]interface{})) {
+		cloudDisableAzureIntegration.AzureVirtualmachine = []cloud.CloudDisableAccountIntegrationInput{{LinkedAccountId: linkedAccountID}}
+	}
+
+	if v, ok := d.GetOk("virtual_networks"); ok {
+		cloudAzureIntegration.AzureVirtualnetworks = expandCloudAzureIntegrationVirtualNetworksInput(v.([]interface{}), linkedAccountID)
+	} else if o, n := d.GetChange("virtual_networks"); len(n.([]interface{})) < len(o.([]interface{})) {
+		cloudDisableAzureIntegration.AzureVirtualnetworks = []cloud.CloudDisableAccountIntegrationInput{{LinkedAccountId: linkedAccountID}}
+	}
+
+	if v, ok := d.GetOk("vms"); ok {
+		cloudAzureIntegration.AzureVms = expandCloudAzureIntegrationVmsInput(v.([]interface{}), linkedAccountID)
+	} else if o, n := d.GetChange("vms"); len(n.([]interface{})) < len(o.([]interface{})) {
+		cloudDisableAzureIntegration.AzureVms = []cloud.CloudDisableAccountIntegrationInput{{LinkedAccountId: linkedAccountID}}
+	}
+
+	if v, ok := d.GetOk("vpn_gateway"); ok {
+		cloudAzureIntegration.AzureVpngateways = expandCloudAzureIntegrationVpnGatewayInput(v.([]interface{}), linkedAccountID)
+	} else if o, n := d.GetChange("vpn_gateway"); len(n.([]interface{})) < len(o.([]interface{})) {
+		cloudDisableAzureIntegration.AzureVpngateways = []cloud.CloudDisableAccountIntegrationInput{{LinkedAccountId: linkedAccountID}}
+	}
+
+	configureInput := cloud.CloudIntegrationsInput{
+		Azure: cloudAzureIntegration,
+	}
+
+	disableInput := cloud.CloudDisableIntegrationsInput{
+		Azure: cloudDisableAzureIntegration,
+	}
+
+	return configureInput, disableInput
+}
+
+// Expanding the AzureAPIManagement
+
+func expandCloudAzureIntegrationAPIManagementInput(b []interface{}, linkedAccountID int) []cloud.CloudAzureAPImanagementIntegrationInput {
+	expanded := make([]cloud.CloudAzureAPImanagementIntegrationInput, len(b))
+
+	for i, azureAPIManagement := range b {
+		var azureAPIManagementInput cloud.CloudAzureAPImanagementIntegrationInput
+
+		if azureAPIManagement == nil {
+			azureAPIManagementInput.LinkedAccountId = linkedAccountID
+			expanded[i] = azureAPIManagementInput
+			return expanded
+		}
+
+		in := azureAPIManagement.(map[string]interface{})
+
+		azureAPIManagementInput.LinkedAccountId = linkedAccountID
+
+		if m, ok := in["metrics_polling_interval"]; ok {
+			azureAPIManagementInput.MetricsPollingInterval = m.(int)
+		}
+		if r, ok := in["resource_groups"]; ok {
```

Comment:
Since `resource_groups` is a list, you'll want to range over it and map all the values. Similar to this 

```
		if r, ok := in["resource_groups"]; ok {
			resourceGroups := r.([]interface{})
			var groups []string

			for _, group := range resourceGroups {
				groups = append(groups, group.(string))
			}
			azureAPIManagementInput.ResourceGroups = groups
		}
```